native_client/javascript/client.ts

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env node
+
+import Ds from "deepspeech";
+import Fs from "fs";
+import Sox from "sox-stream";
+import argparse from "argparse";
+import util from "util";
+
+const MemoryStream = require("memory-stream");
+const Wav = require("node-wav");
+const Duplex = require("stream").Duplex;
+
+console.error('WARNING: THIS IS ONLY INTENDED TO PERFORM CI CHECKS.')
+
+var VersionAction = function VersionAction(options: argparse.ArgumentOptions) {
+  options = options || {};
+  options.nargs = 0;
+  // @ts-ignore: TS2683
```

Comment:
Maybe using the arrow notation will resolve the TS error (See the [TS Handbook](https://www.typescriptlang.org/docs/handbook/functions.html#this))

```
let VersionAction = (options: argparse:ArgumentOptions): void => {
...
}
```

Reply:
This is not helping:
```
TSError: ⨯ Unable to compile TypeScript:
client.ts:18:24 - error TS7041: The containing arrow function captures the global value of 'this'.

18   argparse.Action.call(this, options);
```

Reply:
Also, arrow function breaks `util.inherits`.

Reply:
https://gist.github.com/reuben/2d12285b4de10d7fea89457431128181

Reply:
@reuben I thought about that but I'm unsure whether it will work with NodeJS 10/11/12, so I'll try it after I get something working.

Reply:
It gets compiled down to plain old JS:

```javascript
var VersionAction = /** @class */ (function (_super) {
    __extends(VersionAction, _super);
    function VersionAction() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    VersionAction.prototype.call = function (parser, namespace, values, optionString) {
        console.log('DeepSpeech ' + deepspeech_1["default"].Version());
        console.error('Runtime: Node');
        process.exit(0);
    };
    return VersionAction;
}(argparse.Action));
```

Reply:
Looks like I missed a couple of things: https://gist.github.com/reuben/2d12285b4de10d7fea89457431128181